src/qurveros/spacecurve.py

```diff
@@ -90,35 +92,6 @@ def __init__(self, *, curve, order, interval, params=None,
 
         # Ensure correct types for frenet_dict calculations.
         if curve is not None:
-            if isinstance(curve, str):
```

Comment:
This whole deleted block from line 93 to line 121 should not have been deleted. When you look at the diff between the main branch and your code, you should see only the two insertions in lines 9 and 10, and then the insertions for the type conversion.